Replace plural Alerts with singular Alert.

amrc-benmorrow opened the pull request:
I was trying to micro-optimise and avoid allocating a UUID for every alert. This was an incorrect instinct and causes more problems than it solves. Replace the whole plural schema with a singular schema and fix the references.

amrc-benmorrow committed:
Replace plural Alerts with singular Alert.

I was trying to micro-optimise and avoid allocating a UUID for every
alert. This was an incorrect instinct and causes more problems than it
solves. Replace the whole plural schema with a singular schema and fix
the references.

Commit:
Fix incorrect bracket in Edge_Agent-v1.json

Commit:
Make Type and Active dynamic in Alert-v1.json

This is not ideal but it aligns to the current way of doing things. Ideally we would be able to set these statically per schema. Edge_Agent-v1 still needs reviewing.

AlexGodbehere commented:
@amrc-benmorrow We need to check whether or not we can specify an override for a group of externally referenced schemas. `Edge_Agent-v1` is where this needs to be done in this instance.

We need all alerts within the `Config_Unavailable` metric to have a `Type` hardcoded to `c414c68b-5014-4e46-a0b4-d5f7f8df1d9f`. The manager will still allow users to change this, which isn't ideal so we could do with adding the ability to set fields as readonly to the manager too.

amrc-benmorrow committed:
Don't use JSON schema the Manager doesn't like

The Manager's understanding of JSON Schema is somewhat limited. Restrict
the Edge Agent schema to patterns the Manager will understand.

amrc-benmorrow commented:
I've rebased this onto `main`, and added a commit removing the restrictions on alert type in the Edge_Agent schema.
